docs/VIRTUAL-EVE-DEPLOYMENT.md

```diff
@@ -0,0 +1,268 @@
+* [VMware Fusion](https://www.vmware.com/products/fusion.html)
+* [VMware Workstation Player](https://www.vmware.com/products/workstation-player.html)
+* [VMware Workstation Pro](https://www.vmware.com/products/workstation-pro.html)
+* [Google Cloud Platform (GCP) Compute Engine](https://cloud.google.com/compute)
+
+This document will cover some important notes and instructions for running EVE-OS on the following virtualization platforms:
+1. [VirtualBox and VMware](#deploying-in-virtualbox-and-vmware)
+2. [GCP Compute Engine](#deploying-in-gcp)
+
+## Deploying in VirtualBox and VMware
+
+Prerequisites for deploying EVE-OS in a virtual machine are as follows:
+1. EFI-compatible virtualization platform
+2. A machine with [docker](https://docs.docker.com/engine/install/) installed to generate EVE-OS installer ISO image
+
+Follow these steps to create EVE-OS as a virtual machine:
+1. Create an EVE-OS installer ISO image: ```docker run lfedge/eve:<eve-os-version>-kvm-amd64 installer_iso > installer.iso``` 
+2. Create a Linux VM (Other Linux 64-bit) for EVE-OS VM the with desired CPU, memory and disk
+3. Set firmware of the VM to **EFI**
+4. Attach EVE-OS **installer.iso** ISO image to virtual optical/DVD drive
+5. Change boot order and set DVD drive to boot first
+6. Power on EVE-OS VM and let the installer runs. At the end of its run, the installer process will shut down the EVE-OS VM
+7. Detach installer.iso ISO image from the virtual optical/DVD drive
+8. Power on EVE-OS VM, perform VM console and retrieve serial number from /config/soft_serial to onboard EVE-OS VM to controller: ```cat /config/soft_serial```
+
+
+#### Notes on EFI Firmware
+
+The VM for EVE-OS must have the firmware set to **EFI** in order to boot the ISO installer image. In VirtualBox, firmware settings can be changed from the User Interface or the VBoxManage command line: ```VBoxManage modifyvm [vm-name] --firmware efi```. In VMware (e.g., Fusion, Workstation), you can edit the .vmx file and add the line: ```firmware = "efi"``` or in some version the firmware setting is available under the *Advanced Option*.
+
+References:
+* [Alternative Firmware (EFI) in VirtualBox](https://docs.oracle.com/en/virtualization/virtualbox/6.0/user/efi.html)
+* [Configure a Firmware Type in VMware Workstation Pro](https://docs.vmware.com/en/VMware-Workstation-Pro/16.0/com.vmware.ws.using.doc/GUID-064517C9-14D6-4C87-8D2C-2856EFAE88EB.html)
+
+####Sample VBoxManage script to create an EVE-OS VM
+```sh
+VBoxManage createvm --name [vm-name] --ostype Linux_64 --register
+VBoxManage modifyvm [vm-name] --cpus 1 --memory 1024 --nic1 nat --hwvirtex on --vtxvpid on --firmware efi
+VBoxManage createhd --filename "[path]\[disk-name].vdi" --size [size in MiB units] --format VDI
+VBoxManage storagectl [vm-name] --name "IDE Controller" --add ide --controller PIIX4
+VBoxManage storageattach [vm-name] --storagectl "IDE Controller" --port 0 --device 0 --type hdd --medium "[path]\[disk-name].vdi"
```

Comment:
It's better to use a SATA emulated controller here, SATA emulation has a better performance than IDE emulation on Virtualbox...